### ISSUE TITLE
fb_season_team_stats can't recognise some home/away league tables

### ISSUE THREAD

**Issue by szfh** [JaseZiv/worldfootballR#282]
(reported by email)

I encountered the following error when trying to scrape the data from the Mexican league table, seasons 2007 to 2022.

seasons <- c(2017:2022)
```r
df_leaguetables_MEX <-
  fb_season_team_stats(
    country = "MEX",
    gender = "M",
    season_end_year = seasons,
    tier = "1st",
    stat_type = "league_table_home_away",
    time_pause = 3
  )
```
```
Error in `map()`:
ℹ In index: 1.
Caused by error in `dplyr::mutate()`:
ℹ In argument: `Team_or_Opponent = ifelse(...)`.
Caused by error in `.data[["Squad"]]`:
! Column `Squad` not found in `.data`.
Backtrace:
  1. worldfootballR::fb_season_team_stats(...)
 30. rlang:::abort_data_pronoun(x, call = y)
 ```

**Comment by szfh** [comment]
This is because of the way Mexican leagues are structured.

```r
mex_data_2021 <-
  fb_season_team_stats(
    country = "MEX",
    gender = "M",
    season_end_year = c(2021:2022),
    tier = "1st",
    stat_type = "league_table_home_away",
    time_pause = 3
  )
```
Seasons from 2021 work as expected, because the Overall and Home/Away tabs are the first tables in the page.
https://fbref.com/en/comps/31/2020-2021/2020-2021-Liga-MX-Stats

```r
mex_data2020 <-
  fb_season_team_stats(
    country = "MEX",
    gender = "M",
    season_end_year = 2020,
    tier = "1st",
    stat_type = "league_table_home_away",
    time_pause = 3
  )
```
Seasons 2020 and earlier don't work because the web page is laid out differently with the "main" league table further down the page.
https://fbref.com/en/comps/31/2019-2020/2019-2020-Liga-MX-Stats

**Comment by JaseZiv** [comment]
Thanks for raising here @szfh.

I will look into this over the next little while, unless you wanted to tackle it? Zero stress if not though.

**Comment by szfh** [comment]
I'll have a look and see if I can update it, if not I'll pass it over.

**Comment by szfh** [comment]
@JaseZiv could you do the assignment and labels? I don't have the right permission I think

**Comment by szfh** [comment]
Related issue: for the same reason, this function doesn't work as expected. It gets the first table on the page, which is the 2019 Apertura, not the full season league table.

```r
mex_data_2020 <-
  fb_season_team_stats(
    country = "MEX",
    gender = "M",
    season_end_year = 2020,
    tier = "1st",
    stat_type = "league_table",
    time_pause = 3
  )
```

**Comment by JaseZiv** [comment]
Closed with #285
